components/datetime/src/input.rs

```diff
@@ -344,11 +343,7 @@ fn day_of_week_in_month<T: DateInput>(datetime: &T) -> Result<DayOfWeekInMonth,
 }
 
 impl<'data, T: DateTimeInput> DateTimeInputWithLocale<'data, T> {
-    pub fn new(
-        data: &'data T,
-        calendar: Option<&'data week_of::CalendarInfo>,
-        _locale: &DataLocale,
```

Comment:
Note: This function here is the point at which the locale is simply being ignored.